### ISSUE TITLE
Top 5 Agents not shown anymore under Overview > Integrity monitoring since Wazuh update 3.5 -> 3.6

### ISSUE THREAD

**Issue by juergenc** [wazuh/wazuh-kibana-app#889]
Hello,

I upgraded all components from release 3.5 to 3.6. See here the here on manager host:
```
dpkg --list wazuh-\* elasticsearch logstash kibana
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                  Version                         Architecture                    Description
+++-=====================================================-===============================-===============================-===============================================================================================================
ii  elasticsearch                                         6.4.0                           all                             Elasticsearch is a distributed RESTful search engine built for the cloud. Reference documentation can be found 
ii  kibana                                                6.4.0                           amd64                           Explore and visualize your Elasticsearch data
ii  logstash                                              1:6.4.0-1                       all                             An extensible logging pipeline
un  wazuh-agent                                           <none>                          <none>                          (no description available)
ii  wazuh-api                                             3.6.0-1                         amd64                           Wazuh API is an open source RESTful API to interact with Wazuh from your own application or with a simple web b
ii  wazuh-manager                                         3.6.0-2                         amd64                           Wazuh helps you to gain security visibility into your infrastructure by monitoring hosts at an operating system
```
 and the _Wazuh App_ with _wazuhapp-3.6.0_6.4.0.zip_.

Now, if you open page `Overview > Integrity monitoring` for all agents in _Kibana_ frontend, the pie chart _Top 5 agents_ and the stacked bar graph _Alerts evolution - Top 5 agents_ shows up only _one_ of a total of six agents, just the one running on the manager host. If I go back in the timeline or shift the time window to the past _before_ the upgrade, Kibana shows in the pie chart _Top 5 agents_ and the stacked bar graph _Alerts evolution - Top 5 agents_ indeed five agents. I don't know how to upload screenshots for better explanation. 
It is the same issue for **all** tabs below _Overview_ monitoring for the manager node is shown, but the other agents definitely report. I can see this if I open the _Ageents_ tab, pick an arbitrary agent and chose any monitoring tab, e.g. _Security events_, _Integrity monitoring_, _Policy monitoring_, ...

I don't know where to look here, may be I messed something up when I upgraded to Wazuh 3.6 (probably Wazuh App). Do you have an idea where to look?

Thanks!

**Comment by JuanjiJG** [comment]
Hello @juergenc,

There's a known bug on Wazuh v3.6.0 that removed the `agent.name` field from alerts. That's why you're not seeing your agent names on several visualizations, because they're built using that specific alert field.

I strongly recommend you to upgrade to our latest version, Wazuh v3.6.1, which includes a fix for this, along with several improvements.

Read [this guide](https://documentation.wazuh.com/current/installation-guide/upgrading/latest_wazuh3_minor.html) for more information about how to upgrade.

Let me know if you need more help with this.

Regards,
Juanjo

**Comment by juergenc** [comment]
Hello,

I updated all components on manager where there is the newer release _3.6.1_. Now, the regression bug is fixed! But, alas, I detected another regression which was already filed as a bug and fixed, probably not added to release branch 3.6.1 maybe to a later one. The regression here is that the package _inventory_ for an agents shows up not an error anymore as it was after the upgrade from release 3.5 to 3.6, now from 3.6.0 to 3.6.1 it claims _syscollector_ is not configured which is not the truth. 
When I upgrade from 3.5 to 3.6 and the error message was shown in Kibana when clicked on _inventory_ for a specific agent, I detected there was filed already a bug report (I cannot find it anymore) where the developer exactly reported what is wrong here: 

In line _64_ of Python script `/var/ossec/framework/wazuh/wdb.py` the line 
```Python
data = self.__conn.recv(data_size).decode().split(" ", 1)
```
needs to be replaced by
```Python
data = self.__conn.recv(data_size).decode('utf-8').split(" ", 1)
```
I did so some other folks in the community after the upgrade to 3.6, restarted Kibana service and it worked. Now, after upgrade from 3.6.0 to 3.6.1, line 64 has again the old entry **without UTF-8** decoding:
```Python
data = self.__conn.recv(data_size).decode().split(" ", 1)
```
Again I changed it to 
```Python
data = self.__conn.recv(data_size).decode('utf-8').split(" ", 1)
```
restarted the Kibana service and worked again.

Why was this change not merged into 3.6.1 release branch? It's a bit annoying for that folks who changed it and after the upgrade to 3.6.1 it is reverted back. 

Damn, I cannot find the corresponding bug ticket, the only thing I remember it already closed and I don't know in which branch the fix was merged.

**Comment by jesusgn90** [comment]
Hi @juergenc, 

I'm agree with you. The UTF-8 problem was reported and you can track it following the next ticket https://github.com/wazuh/wazuh-api/issues/150. What happened was you modify your code using the suggestion from one Wazuh team member, but that change was not introduced in any release, once you updated your environment that change was replaced... In any case it will be release in next packages.

Regarding to the documentation for upgrading the Wazuh app, it was changed in the release PR (https://github.com/wazuh/wazuh-documentation/pull/372, [files#diff-fa6e07d6c446af6836948ee00dcf9552L251](https://github.com/wazuh/wazuh-documentation/pull/372/files#diff-fa6e07d6c446af6836948ee00dcf9552L251)).

And thanks about the inventory comment. More changes are coming soon, stay updated. Your feedback is really appreciated.

Best regards,
Jesús

**Comment by juergenc** [comment]
If I see it correctly, the fix will merged into release branch 3.7. For all future releases between 3,6,1 and 3,7.0 people will still experience this issue when they upgrade from 3.5 where inventory tab still works. Alright, to me, I know what I have to do if upgrade next time to a release < 3.7.0.

Thanks for all.
Jürgen

**Comment by jesusgn90** [comment]
Yes @juergenc , 

That's right. And for your information, the next release will be probably v3.7.0, I think there won't be more 3.6.x releases, but it's just an opinion.

Thanks for your patience.

Regards,
Jesús
